govcd/vm_test.go

```diff
@@ -416,3 +416,146 @@ func (vcd *TestVCD) Test_VMDetachDisk(check *C) {
+
+	insertMediaTask, err := vm.HandleInsertMedia(&vcd.org, vcd.config.VCD.Catalog.Name, itemName)
+	check.Assert(err, IsNil)
+
+	err = insertMediaTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//verify
+	err = vm.Refresh()
+	check.Assert(err, IsNil)
+	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, true)
+
+	ejectMediaTask, err := vm.HandleEjectMedia(&vcd.org, vcd.config.VCD.Catalog.Name, itemName)
+	check.Assert(err, IsNil)
+
+	err = ejectMediaTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//verify
+	err = vm.Refresh()
+	check.Assert(err, IsNil)
+	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, false)
+}
+
+// Test Insert or Eject Media for VM
+func (vcd *TestVCD) Test_InsertOrEjectMedia(check *C) {
+
+	itemName := "TestInsertOrEjectMedia"
+
+	// Find VApp
+	if vcd.vapp.VApp == nil {
+		check.Skip("skipping test because no vApp is found")
+	}
+
+	vapp := vcd.find_first_vapp()
+	vmType, vmName := vcd.find_first_vm(vapp)
+	if vmName == "" {
+		check.Skip("skipping test because no VM is found")
+	}
+
+	fmt.Printf("Running: %s\n", check.TestName())
+
+	vm := NewVM(&vcd.client.Client)
+	vm.VM = &vmType
+
+	// Upload Media
+	catalog, err := vcd.org.FindCatalog(vcd.config.VCD.Catalog.Name)
+	check.Assert(err, IsNil)
+
+	uploadTask, err := catalog.UploadMediaImage(itemName, "upload from test", vcd.config.Media.MediaPath, 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	AddToCleanupList(itemName, "mediaImage", vcd.org.Org.Name+"|"+vcd.vdc.Vdc.Name, "Test_UploadMediaImage")
+
+	media, err := FindMediaAsCatalogItem(&vcd.org, vcd.config.VCD.Catalog.Name, itemName)
+	check.Assert(err, IsNil)
+	check.Assert(media, Not(Equals), CatalogItem{})
+
+	// Insert Media
+	insertMediaTask, err := vm.insertOrEjectMedia(&types.MediaInsertOrEjectParams{
+		Media: &types.Reference{
+			HREF: media.CatalogItem.Entity.HREF,
+			Name: media.CatalogItem.Entity.Name,
+			ID:   media.CatalogItem.Entity.ID,
+			Type: media.CatalogItem.Entity.Type,
+		},
+	}, types.RelMediaInsertMedia)
+	check.Assert(err, IsNil)
+
+	err = insertMediaTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//verify
+	err = vm.Refresh()
+	check.Assert(err, IsNil)
+
+	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, true)
+
+	// Insert Media
+	ejectMediaTask, err := vm.insertOrEjectMedia(&types.MediaInsertOrEjectParams{
+		Media: &types.Reference{
+			HREF: media.CatalogItem.Entity.HREF,
+		},
+	}, types.RelMediaEjectMedia)
+	check.Assert(err, IsNil)
+
+	err = ejectMediaTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//verify
+	err = vm.Refresh()
+	check.Assert(err, IsNil)
+	check.Assert(isMediaInjected(vm.VM.VirtualHardwareSection.Item), Equals, false)
+}
+
+func isMediaInjected(items []*types.VirtualHardwareItem) bool {
+	isFound := false
+	for _, hardwareItem := range items {
+		if hardwareItem.ResourceSubType == "vmware.cdrom.iso" {
```

Comment:
Where does this literal value come from? Can use use a constant with a good name and documentation instead?

Reply:
No documentation found - I figure out from responses